File: linkedlist/linkedlist.go
package main

import (
	"log"
)

type Node struct {
	value interface{}
	next *Node
}

type LinkedList struct {
	start *Node
	end *Node
}
	
func (ll *LinkedList) addNode( _value interface{} ) {
	if ll.start == nil {
		ll.start = &Node{value: _value, next:nil}
		ll.end = ll.start
	} else {
		ll.end.next = &Node{value: _value, next:nil}
		ll.end = ll.end.next
	}
}

func (ll *LinkedList) removeNode( ) {
	if ll.end != nil {
		temp := ll.start
		for temp.next != ll.end {
			temp = temp.next
		}
		temp.next = nil
	}
}

func main() {
	var llist LinkedList
	llist.addNode("Banana")
	llist.addNode("Split")
	llist.addNode("and Apple pie")
	llist.removeNode()
	var node = llist.start
	for node != nil {
		log.Print(node.value)
		node = node.next
	}
}

File: linkedlist/linkedlist.scala
// linkedlist.scala

class Node[T](_value: T) {
	var value = _value
	var next : Node[T] = _;
}

class LinkedList[T] {
	var start : Node[T] = _;
	var end : Node[T] = _;
	def addNode(_value: T) = { 
		if ( start == null ) {
			start = new Node[T](_value)
			end = start
		} else {
			end.next = new Node[T](_value)
			end = end.next
		}
	}
	def removeNode() = { 
		if ( end != null ) {
			var temp : Node[T] = start
			while ( temp.next != end ) {
				temp = temp.next
			}
			temp.next = null
		}
	}
}

var llist = new LinkedList[String]
llist.addNode("Banana")
llist.addNode("Split")
llist.addNode("and Apple Pie")
llist.removeNode()

var node = llist.start
while ( node != null ) {
	print(node.value + " ")
	node = node.next
}
println("")
